examples/active-active/main.tf

```diff
@@ -0,0 +1,96 @@
+provider "aws" {
+  default_tags {
+    tags = var.tags
+  }
+  region = "us-east-2"
+  assume_role {
+    role_arn = var.aws_role_arn
+  }
+}
+
+resource "random_string" "friendly_name" {
+  length  = 4
+  upper   = false # Some AWS resources do not accept uppercase characters.
+  number  = false
+  special = false
+}
+
+# Keypair for SSH
+# ---------------
+resource "tls_private_key" "main" {
+  algorithm = "RSA"
+}
+
+resource "local_file" "private_key_pem" {
+  filename = "${path.module}/work/private-key.pem"
+
+  content         = tls_private_key.main.private_key_pem
+  file_permission = "0600"
+}
+
+resource "aws_key_pair" "main" {
+  public_key = tls_private_key.main.public_key_openssh
+
+  key_name = "${local.friendly_name_prefix}-ssh"
+}
+
+# Key Management Service
+# ----------------------
+module "kms" {
```

Comment:
This is another one of those usability vs simplicity things and this being an example might push hard to simplicity:  Most consumers that care about a KMS are, I'd guess, likely to have an existing KMS managed by a team with that KMS being a highly audited function of their infrastructure.  I'd almost want to see a switch in the example to override the kms and data-block the required identifier if such an external KMS existed.

Also this is an example and might not need an extra layer of complexity so I can see both ways here as well.

Reply:
I think you're right, but I think that we should add a separate example with an existing KMS.